stdlib/public/SwiftShims/Visibility.h

```diff
@@ -236,6 +236,11 @@
 #define SWIFT_NODISCARD
 #endif
 
+#if __has_cpp_attribute(gnu::returns_nonnull)
+#define SWIFT_RETURNS_NONNULL [[gnu::returns_nonnull]] SWIFT_NODISCARD
```

Comment:
Seems like `SWIFT_NODISCARD` should be a separate thing. It's not really implied by "returns nonnull."

Reply:
That's fair—my thinking was just in terms of allocation but a function might be nonnull-returning and not actually allocate anything.

Reply:
Right. It definitely makes sense for these particular functions to be nodiscard, but it doesn't go with nonnull in general.